src/notification.c

```diff
@@ -199,19 +199,33 @@ void notification_free(notification * n)
 }
 
 /*
- * Replace all occurrences of "needle" with a quoted "replacement",
- * according to the markup settings.
+ * Replace the two chars where **needle points
+ * with a quoted "replacement", according to the markup settings.
+ *
+ * The needle is a double pointer and gets updated upon return
+ * to point to the first char, which occurs after replacement.
+ *
  */
-char *notification_replace_format(const char *needle, const char *replacement,
-                                  char *haystack, enum markup_mode markup_mode) {
-        char* tmp;
-        char* ret;
+void notification_replace_single_field(char **haystack, char **needle,
+                const char *replacement, enum markup_mode markup_mode) {
 
-        tmp = markup_transform(g_strdup(replacement), markup_mode);
-        ret = string_replace_all(needle, tmp, haystack);
-        g_free(tmp);
+        assert(*needle[0] == '%');
+        // needle has to point into haystack
+        assert(*needle >= *haystack);
+        assert(*needle - *haystack <= strlen(*haystack));
+
+        char* input;
```

Comment:
While currently it's already in a similar way in `master` I think it's worth fixing these nitpicks since we're refactoring this anyway:

1) There is no reason to have this declaration here and not a few lines bellow where `input` is first used
2) While it's pretty inconsistent across the project, we usually put the asterisk next to the variable name and not the type e.g. `char *input`.
The rationale for this is simply that it gets confusing when using commas, for example in `char* a,b` `a` is a pointer while `b` is not.

Reply:
> While it's pretty inconsistent across the project, we usually put the asterisk next to the variable name and not the type e.g. char *input.
The rationale for this is simply that it gets confusing when using commas, for example in char* a,b a is a pointer while b is not.

Thanks for answering this question. I knew this was the C-specific equivalent of "tabs vs spaces", so I kept avoiding it, because I thought I could not understand all arguments. But thanks it's very clear now.

> There is no reason to have this declaration here and not a few lines bellow where input is first used

So, while we're at these "religious" questions:

```
char *string = function_returning_string();
```

```
[begin of the block]
char *string;
[maybe some code]
string = function_returning_string();
```

I've seen both styles here in the project, but I haven't found a scheme. My observations had been, that a C-programmer rather uses the 2nd style.

What's better or when use what?